### ISSUE TITLE
`argparse` replaces `"--"` with `[]` in the argument of an append action

### ISSUE THREAD

**Issue by jakelishman** [python/cpython#100804]
# Bug report

It does not appear to be possible to pass a literal `--` as the argument value in an `append` action.  For example

```python
>>> import argparse
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument("-a", action="append")
>>> parser.parse_args(["-a--"])
Namespace(a=[[]])
```

Since `-a` is "append", I would expect the result to be `Namespace(a=["--"])`.

This seems to be because `ArgumentParser._parse_known_args.<locals>.take_action` uses the top-level `ArgumentParser._get_values` parser to convert the arguments to all actions, which includes the logic to strip `--` from the positional argument sets.  I'm not 100% clear if this a bug or intended behaviour, but it's certainly very surprising to me (and seems to break the `build` package's expectations) that the argument can come out as a list, not a string.

If this is a bug, I think it's a low-priority edge case. My original motivation for trying this was misguided; I was trying to pass a `--` in a list of build options with `--config-settings` in `build` and `pip wheel`, but had forgotten to prefix each with `--build-option=`.

# Your environment

- CPython versions tested on: 3.10.6, 3.11.0
- Operating system and architecture: macOS x64

### Linked PRs
* gh-106586
* gh-106590
* gh-106594

**Comment by timonviola** [comment]
Hello, I think this works as it should and can be closed:
```python
Python 3.14.0a0 (heads/main:dc03ce797a, Jul 13 2024, 10:36:06) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import argparse
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument("-a", action="append")
_AppendAction(option_strings=['-a'], dest='a', nargs=None, const=None, default=None, type=None, choices=None, required=False, help=None, metavar=None, deprecated=False)
>>> parser.parse_args(["-a--"])
Namespace(a=['--'])
```

**Comment by savannahostrowski** [comment]
This was, in fact, fixed with e1976399cd08a0aadeb094cee46b4ebf468fecd9, so 3.11+ works as expected 🎉 !
